File: pascal-triangle/PascalTriangle.java
public class Solution{
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        if (numRows == 0) {
            return res;
        }
        for (int i = 0; i < numRows; ++i) {
            if (i == 0) {
                List<Integer> list = new ArrayList<>();
                list.add(1);
                res.add(list);
            }
            else {
                List<Integer> pre = res.get(res.size() - 1);
                List<Integer> list = new ArrayList<>();
                list.add(pre.get(0));
                for (int j = 1; j < pre.size(); ++j) {
                    int tmp = pre.get(j - 1) + pre.get(j);
                    list.add(tmp);
                }
                list.add(pre.get(pre.size() - 1));
                res.add(list);
            }
        }
        return res;
    }
}

File: pascal-triangle/PascalTriangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        res = []
        if numRows == 0:
            return res
        
        for i in range(numRows):
            if i == 0:
                list = [1]
                res.append(list)
            else:
                pre = res[len(res) - 1]
                list = []
                list.append(pre[0])
                for j in range(1, len(pre)):
                    tmp = pre[j - 1] + pre[j]
                    list.append(tmp)
                list.append(pre[len(pre) - 1])
                res.append(list)
        return res